### ISSUE TITLE
Access to raw DOS filenames and VTOC?

### ISSUE THREAD

**Issue by markdavidlong** [fadden/ciderpress#46]
As a 3rd-party user of the diskimg library, I think it would be handy to have access to the raw DOS filename in order to handle control characters, inverse/flash characters, etc.    Also, it would be convenient to have a way to override the default VTOC Track and Sector constants.  Perhaps some of the DiskFSDOS33 members would be better as protected rather than private, as a subclass might be useful downstream, as well.

**Comment by fadden** [comment]
I thought I'd made the "raw" filename available, but apparently I just [thought about it](https://github.com/fadden/ciderpress/blob/7c637c4e6d2e0f7e0a60676bbda1bcd4618e1234/diskimg/DiskImgDetail.h#L1414).

Apparently I also [thought](https://github.com/fadden/ciderpress/blob/7c637c4e6d2e0f7e0a60676bbda1bcd4618e1234/diskimg/DOS33.cpp#L26) about overriding the T=17 S=0 constants.

It appears that me from 20 years ago agrees with you. :-)

**Comment by markdavidlong** [comment]
Would it be better to have the raw name functionality as a DOS33 (et al.) method, which would require a cast from A2File, or would it be better served as a virtual method in A2File, where the default behavior would be to just return the results of a call to GetFileName()?

**Comment by fadden** [comment]
I figure returning the raw un-fixed name potentially has value on other filesystems, so it'd be part of A2File.

**Comment by markdavidlong** [comment]
I submitted a pull request which, I believe, handles the filename issue, although not on HFS so far.

**Comment by fadden** [comment]
Got it.  I'll take a look soon.  What's the issue with HFS?

**Comment by markdavidlong** [comment]
I'm just fairly unfamiliar with the file system and wasn't confident about making a change without a way to test it on my end. With some trepidation about the HFS code I've reviewed, I don't want to make assumptions about how names are propagated and assigned.  Unlike the DOS file systems which are fairly straightforward, I believe that the key to accommodating nulls in any HFS filenames comes down to preserving a length from element 0 of a Pascal string, but I don't want to introduce any bugs with the myriad places file names seem to be dealt with. As it stands with the code I have in place, the caveat is if an HFS filename is abbreviated due to an embedded null, the length variable will still be in agreement with the (incorrect) "raw" name, at least.

Short version: I could use a little guidance or secondary experience in order to make the HFS raw name code bulletproof, though there is a hook in the new method which can be used to work around an early-terminated string at some point by returning the reported Pascal string length.

**Comment by fadden** [comment]
As far as I can tell, from [Inside Mac: Files chapter 2](http://www.dubeyko.com/development/FileSystems/HFS/inside_macintosh/File_Manager.pdf), page 2-71, HFS filenames are stored in a fixed 31-byte field that is padded with zeroes.  This would mean that it's not possible to have an embedded null in an HFS filename.

The [DiskCopy format](http://nulib.com/library/FTN.e00005.htm) uses a Pascal string, and according to my comments there's some weirdness with volume names, but I don't think any of that is strictly driven by HFS.  Might have been excessive generality on the part of the format designer, might have been planning for a post-HFS filesystem.

I don't know if any of this affects the changes you made -- having an explicit length for a raw filename makes sense -- but I don't think you need to worry about breaking HFS.

FWIW, the CD images for Opus ][ (Byteworks sources) and A2 Romulan are in HFS format.  If you have those you can exercise code with a large HFS volume.
